src/pages/Flats/FlatEditor.tsx

```diff
@@ -1,11 +1,39 @@
-import {useParams} from "react-router-dom";
+import {useLocation} from "react-router-dom";
+import Button from "../../components/utils/Button";
+import {useNavigate} from "react-router-dom";
+import FlatForm from "../../components/forms/FlatForm";
+import {Flat} from "../../common/types/Flat";
+import generateID from "../../common/helpers/generateID";
 
-function FlatEditor() {
-    const { id } = useParams();
+interface FlatEditorParams {
+    mode: 'edit' | 'add'
+}
+
+function FlatEditor(props: FlatEditorParams) {
+    const navigate = useNavigate();
+    const { state } = useLocation();
```

Comment:
Nie przekazywałbym tego w ten sposób, bo wtedy nie możesz komuś podać bezpośredniego linka do edycji.
Lepszym sposobem jest pobranie id z url i wysłanie request GET flats/{id} żeby pobrać info. Wtedy też możesz zrobić catch na wyjątek i zrobić redirect do widoku listy jeśli takiego id nie znajdzie (zobacz jak to jest zrobione przy szczegółach). Wtedy ta strona będzie bardziej niezależna od listy bo teraz to jedyny sposób zeby dostać się do edycji po ten obiekt Flat przekazywany jest przez react-rotuer.

Reply:
done